numpy/_core/src/multiarray/nditer_api.c

```diff
@@ -1340,75 +1339,16 @@ NPY_NO_EXPORT void
 
     if (itflags&NPY_ITFLAG_BUFFER) {
-        NpyIter_BufferData *data = NIT_BUFFERDATA(iter);
-        npyiter_opitflags *op_itflags = NIT_OPITFLAGS(iter);
-        npy_intp stride, *strides = NBF_STRIDES(data),
-                *ad_strides = NAD_STRIDES(axisdata0);
-        PyArray_Descr **dtypes = NIT_DTYPES(iter);
-
-        for (iop = 0; iop < nop; ++iop) {
-            stride = strides[iop];
-            /*
-             * Operands which are always/never buffered have fixed strides,
-             * and everything has fixed strides when ndim is 0 or 1
-             */
-            if (ndim <= 1 || (op_itflags[iop]&
-                            (NPY_OP_ITFLAG_CAST|NPY_OP_ITFLAG_BUFNEVER))) {
-                out_strides[iop] = stride;
-            }
-            /* If it's a reduction, 0-stride inner loop may have fixed stride */
-            else if (stride == 0 && (itflags&NPY_ITFLAG_REDUCE)) {
-                /* If it's a reduction operand, definitely fixed stride */
-                if (op_itflags[iop]&NPY_OP_ITFLAG_REDUCE) {
-                    out_strides[iop] = stride;
-                }
-                /*
-                 * Otherwise it's guaranteed to be a fixed stride if the
-                 * stride is 0 for all the dimensions.
-                 */
-                else {
-                    NpyIter_AxisData *axisdata = axisdata0;
-                    int idim;
-                    for (idim = 0; idim < ndim; ++idim) {
-                        if (NAD_STRIDES(axisdata)[iop] != 0) {
-                            break;
-                        }
-                        NIT_ADVANCE_AXISDATA(axisdata, 1);
-                    }
-                    /* If all the strides were 0, the stride won't change */
-                    if (idim == ndim) {
-                        out_strides[iop] = stride;
-                    }
-                    else {
-                        out_strides[iop] = NPY_MAX_INTP;
-                    }
-                }
-            }
-            /*
-             * Inner loop contiguous array means its stride won't change when
-             * switching between buffering and not buffering
-             */
-            else if (ad_strides[iop] == dtypes[iop]->elsize) {
-                out_strides[iop] = ad_strides[iop];
-            }
-            /*
-             * Otherwise the strides can change if the operand is sometimes
-             * buffered, sometimes not.
-             */
-            else {
-                out_strides[iop] = NPY_MAX_INTP;
-            }
-        }
+        /* If there is buffering we wrote the strides into the bufferdata. */
+        memcpy(out_strides, NBF_STRIDES(NIT_BUFFERDATA(iter)), nop*NPY_SIZEOF_INTP);
     }
     else {
-        /* If there's no buffering, the strides are always fixed */
+        /* If there's no buffering, the strides come from the operands. */
         memcpy(out_strides, NAD_STRIDES(axisdata0), nop*NPY_SIZEOF_INTP);
```

Comment:
A future cleanup might  eliminate any internal calls to this API function and use the iter data directly.